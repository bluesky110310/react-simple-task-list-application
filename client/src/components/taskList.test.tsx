import { cleanup } from '@testing-library/react';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import TaskList from './taskList';
import { configure, shallow } from 'enzyme';
import { AppBar, Button, Stack, Typography } from '@mui/material';

afterEach(() => {
    cleanup();
});

configure({ adapter: new Adapter() });

const setup = () => {
    const wrapper = shallow(<TaskList />);

    return { wrapper };
}

describe('TaskList component', () => {
    test('it renders', () => {
        const { wrapper } = setup();

        expect(wrapper).toMatchSnapshot();

        const title = wrapper.find(AppBar).find(Typography);
        expect(title.at(0).text()).toEqual('Task List');

        const button = wrapper.find(Stack).find(Button);
        expect(button.at(0).text()).toEqual('Add task');
    });
});