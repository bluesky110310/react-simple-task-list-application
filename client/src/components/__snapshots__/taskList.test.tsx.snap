// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList component it renders 1`] = `ShallowWrapper {}`;
